src/compiler/checker.ts

```diff
@@ -18293,6 +18293,13 @@ namespace ts {
                 error(indexExpression, Diagnostics.A_const_enum_member_can_only_be_accessed_using_a_string_literal);
                 return errorType;
             }
+            if (isTupleType(objectType) && !objectType.target.hasRestElement && isNumericLiteral(indexExpression)) {
+                const index = +indexExpression.text;
+                const maximumIndex = length(objectType.target.typeParameters) - 1;

```

Comment:
Sorry, I had to remove my comments as soon as @RyanCavanaugh proposed to just change error message text (which I think better solution - it handled cases with empty tuples well without special treatment).
With new message - changes that I proposed earlier only [break stuff](https://github.com/Microsoft/TypeScript/pull/26292/files#diff-e03d8ed536fc4ddafec16efc8d311149R8) 😞  and should be reverted.